desktop/packages/linux/rs.ruffle.Ruffle.metainfo.xml

```diff
@@ -0,0 +1,73 @@
+  <id>rs.ruffle.Ruffle</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Ruffle</name>
+  <summary>A Flash Player emulator written in Rust</summary>
+  <launchable type="desktop-id">rs.ruffle.Ruffle.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/ruffle-rs/ruffle/master/desktop/packages/linux/launcher.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/ruffle-rs/ruffle/master/desktop/packages/linux/btd.png</image>
+    </screenshot>
+    <screenshot>
+      <image>https://raw.githubusercontent.com/ruffle-rs/ruffle/master/desktop/packages/linux/ltf.png</image>
+    </screenshot>
+  </screenshots>
+  <description>
+    <p>
+    Ruffle is a Flash Player emulator written in Rust. Ruffle runs natively on all modern operating systems as a standalone application, and on all modern browsers through the use of WebAssembly. Leveraging the safety of the modern browser sandbox and the memory safety guarantees of Rust, we can confidently avoid all the security pitfalls that Flash had a reputation for. Ruffle puts Flash back on the web, where it belongs - including browsers on iOS and Android!
+    </p>
+    <p>
+    Designed to be easy to use and install, users or website owners may install the web version of Ruffle and existing flash content will "just work", with no extra configuration required. Ruffle will detect all existing Flash content on a website and automatically "polyfill" it into a Ruffle player, allowing seamless and transparent upgrading of websites that still rely on Flash content.
+    </p>
+    <p>
+     Ruffle is an entirely open source project maintained by volunteers. We're all passionate about the preservation of internet history, and we were drawn to working on this project to help preserve the many websites and plethora of content that will no longer be accessible when users can no longer run the official Flash Player. If you would like to help support this project, we welcome all contributions of any kind - even if it's just playing some old games and seeing how well they run.
+    </p>
+  </description>
+  <categories>
+    <category>Game</category>
+    <category>Utility</category>
+    <category>AudioVideo</category>
+  </categories>
+  <keywords>
+    <keyword>flash</keyword>
+    <keyword>swf</keyword>
+  </keywords>
+  <url type="homepage">https://ruffle.rs</url>
+  <url type="bugtracker">https://github.com/ruffle-rs/ruffle/issues</url>
+  <url type="faq">https://discord.gg/ruffle</url>
+  <url type="help">https://discord.gg/ruffle</url>
+  <url type="donation">https://opencollective.com/ruffle</url>
+  <url type="contact">https://discord.gg/ruffle</url>
+  <url type="vcs-browser">https://github.com/ruffle-rs/ruffle</url>
+  <url type="contribute">https://github.com/ruffle-rs/ruffle/blob/master/CONTRIBUTING.md</url>
+  <releases>
+    <release version="nightly-2023-05-26" date="2022-05-26"/>
```

Comment:
To be clear, is this version string supposed to be updated with every nightly release? If so, it may be a good idea to somehow add doing so to CI before this gets merged.

Reply:
This file should be updated with each release of Ruffle with a basic version and date tags.

But I apologize, I am not familiar with github workflows enough to understand how it can be done.

The basic idea would be to hook onto `release-nightly` that commits the repo with the updated metainfo before doing the release process.

Reply:
I'm somewhat against jamming a commit into `master` every single day just for this.
Could this be omitted here?
Perhaps maintained in a separate branch?

Reply:
Standard practice would be updating this file with a changelog. That way users can see in GUI why there is an update and what changes justifying the update. Currently nightlies do not include a changelog, although it would be nice to have one generated so both the metainfo and the github release would have the nightly.

I understand concerns in regards to having this in the main repository. And yes it could be handled in a separate upstream repository, a "ruffle-meta" repository perchance. But then how is versioning handled, syncing the meta repository releases with the main repositories releases?

I also would like to hear feedback from @relrelb who worked on the [AUR package](https://aur.archlinux.org/packages/ruffle-nightly-bin) who also requires the metainfo.

Reply:
As for the changelog, at least GitHub generates one for the nightly releases: https://github.com/ruffle-rs/ruffle/releases

Reply:
> As for the changelog, at least GitHub generates one for the nightly releases: https://github.com/ruffle-rs/ruffle/releases

Apologies, I must have been mixed up and was thinking of another project.

Reply:
While the expected treatment is to update metainfo with each release. I suppose that is a bit too much for the nightly format.

We can skip over having to update the metainfo until Ruffle stable releases are created.

@relrelb Does that sound good for you?

Reply:
Another option could be to generate this based on the tag at build time by for example having some easily replaceable pattern.

Reply:
I don't believe every nightly version should be added here, it's not intended for testing or development.

Only stable versions that need to be published should be added.

I wish ruffle-rs had some kind of version, even if it was alpha for example instead of just nightly builds.

Reply:
We've long planned to come up with something like that, but there's just no clear point of "when" a release should be cut. Every nightly is basically like the next - with a few exceptions one way or another.

That's why the browser extensions and the Android app are "randomly" published: weekly, or every few days/weeks, basically at random.

The same thing could work here, don't you think?

Reply:
I think as long as this hardcoded version here is replaced with some sort of `@PLACEHOLDER@`, to be substituted just before publishing, this should be fine.